ignite/metrics/mean_pairwise_distance.py

```diff
@@ -21,26 +21,26 @@ def __init__(
         p: int = 2,
         eps: float = 1e-6,
         output_transform: Callable = lambda x: x,
-        device: Optional[Union[str, torch.device]] = None,
+        device: Union[str, torch.device] = torch.device("cpu"),
     ):
         super(MeanPairwiseDistance, self).__init__(output_transform, device=device)
         self._p = p
         self._eps = eps
 
     @reinit__is_reduced
     def reset(self):
-        self._sum_of_distances = 0.0
+        self._sum_of_distances = torch.tensor(0.0, device=self._device)
         self._num_examples = 0
 
     @reinit__is_reduced
     def update(self, output: Sequence[torch.Tensor]) -> None:
         y_pred, y = output
         distances = pairwise_distance(y_pred, y, p=self._p, eps=self._eps)
-        self._sum_of_distances += torch.sum(distances).item()
+        self._sum_of_distances += torch.sum(distances).detach().to(self._device)
```

Comment:
Same question than above.

Reply:
Here maybe we can even detach `y_pred` before calling `pairwise_distance`...

Reply:
I agree. If we detach `y_pred`, I think we should also detach `y`, since the user may backprop through it (e.g. for adversarial learning).

Reply:
@n2cholas I think when we do an adversarial learning the target is already "detached" as we do not want to update model's parameters that generated `y`... But it does not cost too much to apply it here though

Reply:
Fixed this in 6edd30d.